db/migrate/20210602211053_create_file_resource_category_relations.rb

```diff
@@ -0,0 +1,10 @@
+class CreateFileResourceCategoryRelations < ActiveRecord::Migration[6.1]
+  def change
+    create_table :file_resource_category_relations do |t|
+      t.bigint :file_resource_id
+      t.bigint :category_id
+
+      t.timestamps
+    end
+  end
+end
```

Comment:
Das ist ein Nachbau von Rails-Feature.  Schaut bitte euch dazu https://edgeguides.rubyonrails.org/active_storage_overview.html an. 🥇 

Damit kannst du beide Ressourcen und Migration ersetzen: `CreateFileResourceCategoryRelations` und `CreateFileResources `

Reply:
Übrigens: als Name dann in diesen Fall nicht sowas wie `FileResource`, sondern einfach nur `file` benutzen. Da in Rails alles eine Ressource ist, was von Rails verwaltet wird. 

Besser aber noch, wenn statt einfach nur `file` aus den Namen klar ersichtlich wird, wozu die Ressource benutzt wird.

Reply:
> Das ist ein Nachbau von Rails-Feature. Schaut bitte euch dazu https://edgeguides.rubyonrails.org/active_storage_overview.html an. 1st_place_medal

ah, ich habe nach dem Feature gesucht gehabt, hatte es aber nicht gefunden

> Übrigens: als Name dann in diesen Fall nicht sowas wie FileResource, sondern einfach nur file benutzen.

ich habe mal am anfang versucht "file" als name zu verwenden, da ist mir aber damals rails zwischen gesprungen. Ich ging davon aus, der hat den namen schon irgendwo für sich selbst eingenommen, höchstwarscheinlich, weil es schon eine "file-class" in rails gibt.

Reply:
Ja, File ist schon als Klasse in Ruby vergeben. Deshalb geht es nicht. Aber, `file` als Active Storage Ressource an einem beliebigen Objekt in Rails, das geht wieder wunderbar. Wobei `file` selbst kein guter Name für eine Ressource ist, wenn es nicht wirklich beliebiger File sein soll.

Reply:
doch, es soll zum schluss ein beliebiges file sein, dazu kommt mehr im nächsten Feature

Reply:
jop, nächstes feature
vielen dank